openquake/hazardlib/source/simple_fault.py

```diff
@@ -160,7 +181,8 @@ def iter_ruptures(self):
                     mesh = whole_fault_mesh[first_row: first_row + rup_rows,
                                             first_col: first_col + rup_cols]
 
-                    if self.hypo_list.size == 1:
+                    if ((self.hypo_list.size == 1) and
```

Comment:
Why when you have only one value on the hypo_list you fix as hypocentre the centroid of the rupture and you don't set it according to the values in the hypo_list?